include/ethernetif.h

```diff
@@ -24,6 +24,7 @@
 
 /* Exported types ------------------------------------------------------------*/
 err_t ethernetif_init(struct netif *netif);
-void ethernetif_input(void *argument);
+void netconfig_init();
```

Comment:
Three things:
1. This should be removed as it's out of scope for `ethernetif`
2. The function declaration is incorrect it should be `void netconfig_init(void)`
3. This declaration should be in `app_ethernet.h`